src/OrbitGl/Batcher.cpp

```diff
@@ -220,15 +220,43 @@ void Batcher::AddTriangle(const Triangle& triangle, const Color& color,
 
 void Batcher::AddTriangle(const Triangle& triangle, const Color& color, const Color& picking_color,
                           std::unique_ptr<PickingUserData> user_data) {
+  std::array<Color, 3> colors;
+  colors.fill(color);
+  AddTriangle(triangle, colors, picking_color, std::move(user_data));
+}
+
+void Batcher::AddShadedTrapezium(const Vec3& top_left, const Vec3& bottom_left,
+                                 const Vec3& bottom_right, const Vec3& top_right,
+                                 const Color& color, std::unique_ptr<PickingUserData> user_data,
+                                 ShadingDirection shading_direction) {
+  std::array<Color, 4> colors;  // top_left, bottom_left, bottom_right, top_right.
+  GetBoxGradientColors(color, &colors, shading_direction);  // left points will be "darker".
+  Color picking_color = PickingId::ToColor(PickingType::kTriangle, user_data_.size(), batcher_id_);
+  Triangle triangle_1{top_left, bottom_left, top_right};
+  std::array<Color, 3> colors_1;
```

Comment:
std::array<Color, 3> colors_1 = {colors[0], colors[1], colors[2]};?

I don't remember the syntax :)

Reply:
Done.